cg_lims/EPPs/udf/copy/original_well_to_sample.py

```diff
@@ -0,0 +1,80 @@
+from email.encoders import encode_quopri
+import logging
+import sys
+import click
+
+from typing import List
+from genologics.entities import Process, Artifact
+from genologics.lims import Lims
+
+from cg_lims import options
+from cg_lims.exceptions import LimsError, InvalidValueError
+from cg_lims.get.artifacts import get_artifacts
+
+LOG = logging.getLogger(__name__)
+
+def original_well_to_sample(
+    artifacts: list
+) -> None:
+    """Function to copy artifact udf location and set to sample level.
+
+    For each artifact in the artifacts list, copying the location and set
+    sample udf 'Original Well' and 'Original Container' to all samples 
+    related to the artifact. 
+
+    Arguments:
+        artifacts: list of artifacts to copy from"""
+
+    failed = 0
+    passed = 0
+    for artifact in artifacts:        
+        if artifact.parent_process:
+            LOG.error(
+            f"Error: Not the first step for sample {artifact.id}. Can therefor not get the original container."
+            )
+            failed += 1
+            continue
+
+        elif len(artifact.samples)!=1:
+            LOG.error(
+                f"Error: more than one sample per artifact for {artifact.id}. Assumes a 1-1 relation between sample and artifact."
+            )
+            failed += 1
+            continue
+
+        for sample in artifact.samples:
```

Comment:
you can skip this loop. There is only going to be one sample according to line 38

```suggestion
sample = artifact.samples[0]
```